src/Resources/Resources/Models.Authorization/AuthorizationClient.cs

```diff
@@ -209,14 +222,35 @@ public List<PSRoleAssignment> FilterRoleAssignments(FilterRoleAssignmentsOptions
                     }
 
                     principalId = adObject.Id.ToString();
-                    odataQuery = new ODataQuery<RoleAssignmentFilter>(f => f.AssignedTo(principalId));
+                    if (!hasScope)
```

Comment:
this conditional is hard to read. hasScope is already a negated statement (scope is NOT null or empty)
remove the ! and swap the code blocks. Makes the flow easier to follow.